docs/cep/stream-worker-ml.md

```diff
@@ -0,0 +1,155 @@
+---
+sidebar_position: 130
+title: Machine Learning with Stream Workers
+---
+
+# Real-time Machine Learning with Macrometa Stream Workers
+
+Macrometa's stream workers offer powerful real-time machine learning capabilities that can be applied to a wide range of use cases. This documentation provides examples with code blocks that showcase the machine learning process. The goal is to help you better understand the application of these functions in real-world scenarios.
+
+For more technical information about Macrometa's machine learning functions, refer to [Streaming ML](query-guide/functions/streaming-ml/index.md).
+
+## Importance of Real-time Machine Learning
+
+Real-time machine learning is crucial for businesses that need to analyze and react to data as it is generated. By processing and learning from data in real-time, you can gain actionable insights faster, make better decisions, and improve the efficiency of your operations. Some of the benefits of real-time machine learning include:
+
+- **Faster decision-making**: Real-time machine learning allows you to make data-driven decisions quickly, giving you a competitive advantage in fast-paced industries.
+- **Enhanced customer experience**: By analyzing customer interactions and behavior in real-time, you can provide personalized experiences and recommendations, leading to higher customer satisfaction and loyalty.
+- **Improved operational efficiency**: Real-time machine learning can help you identify inefficiencies, optimize resource allocation, and reduce costs.
+- **Proactive anomaly detection**: Detecting anomalies in real-time enables you to address potential issues before they escalate, minimizing the impact on your business.
+
+## Example 1: Bayesian Regression for Real-time Energy Consumption Prediction
+
+```sql
+CREATE STREAM TrainingStream (temperature double, time_of_day double, day_of_week double, recent_consumption double, energy_consumption double);
+CREATE STREAM PredictionStream (temperature double, time_of_day double, day_of_week double, recent_consumption double);
+
+CREATE SINK STREAM EnergyConsumptionPredictions (prediction double, confidence double, temperature double, time_of_day double, day_of_week double, recent_consumption double);
+
+@info(name = 'trainEnergyModel')
+INSERT INTO OutputStream1
```

Comment:
@diana-macrometa - `OutputStream1` is an internal stream. Such streams are used for intermediate data processing and finally have to transfer its data to either `sink` or `store`. 
Might be `OutputStream1` should be replaced by `PredictionStream` preliminary defined as `sink` . Otherwise `PredictionStream` is not used at all.
it's the same within the other samples below.